### PR TITLE
Migrate base image to BCI image

### PR DESCRIPTION
**Pull request by bk201**
**Related issue**

https://github.com/harvester/harvester/issues/1868

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,11 @@
-FROM alpine
+FROM registry.suse.com/suse/sle15:15.3
+
 # parted -> for `parted` command
-# util-linux -> for `lsblk` command
+# util-linux-systemd -> for `lsblk` command
 # e2fsprogs -> for `mkfs.ext4` command
-RUN apk update && apk add -u --no-cache parted util-linux e2fsprogs
+RUN zypper -n rm container-suseconnect && \
+    zypper -n install parted util-linux-systemd e2fsprogs && \
```
Comment (weihanglo) [inline package/Dockerfile:7]: Need add `iproute2` package. (Dependency of go-iscsi-helper)
https://software.opensuse.org/package/iproute2

https://github.com/longhorn/go-iscsi-helper/blob/49a327fb024e4637ccc89ec6d73b3c550feec2d1/util/util.go#L94-L96

**Review by weihanglo — COMMENTED** [review]
Note that the image size increases from 48MiB to 200MiB.

**Comment by bk201** [comment]
> Note that the image size increases from 48MiB to 200MiB.

That's mostly caused by the dependency of package `util-linux-systemd`, in alpine world it's more simpler.

File: package/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,11 @@
-FROM alpine
+FROM registry.suse.com/bci/bci-base:15.3
```
Comment (bk201) [inline package/Dockerfile:1]: Also switch from
```
FROM registry.suse.com/suse/sle15:15.3
```

to

```
FROM registry.suse.com/bci/bci-base:15.3
```

**Review by weihanglo — APPROVED** [review]
LGTM

**Comment by bk201** [comment]
FWIW, using dive to inspect image layers:

original (alpine):
Screen Shot 2022-02-15 at 2 46 04 PM

new (BCI):

Screen Shot 2022-02-15 at 2 46 08 PM


The zstd-compressed tarball is 16M vs 63M.

**Comment by guangbochen** [comment]
Wondering if `bci-base` is need for the NDM or using the `bci-minimal` is sufficient enough?

**Comment by bk201** [comment]
@guangbochen  `zypper` is only available in the `bci-base` image. If we want a managed way to install dependent packages (like lsblk), the `bci-base` image is still preferred.

**Review by guangbochen — APPROVED** [review]
LGTM

**Pull request merged**